### ISSUE TITLE
how to get the distance from the center of vehicle to the edge of road

### ISSUE THREAD

**Issue by tinmodeHuang** [carla-simulator/carla#3651]
for that I'm trying to create a sensor to do it as tutorial given to say, but it is too hard for me, cause I'm poor in c++ programm and isn't familiar with UE4 . As an aside, the LaneInvasionSensor cannot detect the action that ego vehicle moves over lanemarkings in time

![white_side](https://user-images.githubusercontent.com/65896325/100710974-88c6e780-33eb-11eb-9f42-e42eca774c4c.gif)

@XGodina I appreciate it if you could give me a litte favor

**Comment by tinmodeHuang** [comment]
@jackbart94 if convient, it is the best that you could help me implement it, which also gives users more material to reference, but that seems impossible, if that way, I wonder how to acquire the edge of road

**Comment by jackbart94** [comment]
Well, the LaneInvasionSensor is a bit broken, so this is what you can do:

- use the get_waypoint() in the location of the vehicle, setting the project_to_road bool to False. This will make sure that you get it at the center of the vehicle.
- then, you want to again use get_waypoint() to find the closest lane marking, and then compute the distance between the two.

You might want to check the [Python API information and code snippet in the documentation](https://carla.readthedocs.io/en/latest/python_api/#getters_9) to do this. There's all the information you need.

**Comment by GyChou** [comment]
Hi,

I found that when the ego vehicle enters into a junction, the yaw of the current waypoint does not match the direction of the target lane, for example, I plan the route to turn left, but the yaw of the waypoint points to straight ahead.

The way I get the current waypoint is:
world.get_map().get_waypoint(vehicle.get_location())

I doubt that this method: world.get_map().get_waypoint(vehicle.get_location()) is to get the nearest waypoint of vehicle.get_location(), due to the waypoints of each direction of the junction are very close, resulting in the selection of the wrong lane waypoint.

I holp get the waypoint of my current location in a junction and its yaw matches my target lane, how do I do that?

I used Carla0.9.11.

Thanks for your help
